src/liballoc/heap.rs

```diff
@@ -16,6 +16,18 @@
 
 use core::{isize, usize};
 
+#[allow(improper_ctypes)]
+extern {
+    #[allocator]
```

Comment:
Currently, yeah, I think that's how it'll work out. We can probably rename the `#[allocator]` attribute on this function to something like `#[noalias]` as that's the name of the actual LLVM attribute anyway.